enterprise/tailnet/pgcoord_test.go

```diff
@@ -415,6 +416,55 @@
 	assertEventuallyLost(ctx, t, store, client.id)
 }
 
+func TestPGCoordinatorSingle_MissedHeartbeats_NoDrop(t *testing.T) {
+	t.Parallel()
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("test only with postgres")
+	}
+	store, ps := dbtestutil.NewDB(t)
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitSuperLong)
+	defer cancel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+
+	coordinator, err := tailnet.NewPGCoord(ctx, logger, ps, store)
+	require.NoError(t, err)
+	defer coordinator.Close()
+
+	agent := test.NewPeer(ctx, t, coordinator, "agent")
+	defer agent.Close(ctx)
```

Comment:
I think the agent on the real coordinator is superfluous to this test.  Just need to test the agent update on the fake coordinator, then have it miss heartbeats.

Reply:
Ah, good point.